plugin/src/main/java/com/yelp/codegen/KotlinGenerator.kt

```diff
@@ -142,10 +142,27 @@ class KotlinGenerator : SharedCodegen() {
     public override fun listTypeWrapper(listType: String, innerType: String) =
             "$listType<$innerType>"
 
+    public override fun listTypeUnwrapper(baseType: String) = baseType
+            .replace(" ", "")
+            .removeSuffix(">")
+            .removePrefix("List<")
+            .removePrefix("Array<")
+
+    public override fun isListTypeWrapped(baseType: String) =
+            baseType.endsWith('>') && (baseType.startsWith("Array<") || baseType.startsWith("List<"))
```

Comment:
As before: `Array` should not be possible and I would replace List with `kotlinTypeMapping["list"]`